# HG changeset patch
# User Pantxo <pantxo.diribarne@gmail.com>
# Date 1429286327 25200
#      Fri Apr 17 08:58:47 2015 -0700
# Node ID ade6e5ae61644f7afc7ca8865a2ec6eb2fc8a58d
# Parent  d6e523825abc61bb6981ba95d20738f52df7ed60
Stop unexpected switches between graphic toolkits (bug #44838).

* graphics.cc (gh_manager::do_make_figure_handle): After making default figure,
use override_defaults to update __graphics_toolkit__ property with current
toolkit.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -9298,16 +9298,18 @@ gh_manager::do_make_figure_handle (doubl
   graphics_object obj (go);
 
   handle_map[h] = obj;
 
   // Notify graphics toolkit.
   if (do_notify_toolkit)
     obj.initialize ();
 
+  obj.override_defaults ();
+
   return h;
 }
 
 void
 gh_manager::do_push_figure (const graphics_handle& h)
 {
   do_pop_figure (h);
 
